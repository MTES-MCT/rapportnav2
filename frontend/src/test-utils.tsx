import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
