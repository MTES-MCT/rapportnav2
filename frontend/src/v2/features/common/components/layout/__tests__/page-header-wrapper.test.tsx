import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import PageHeaderWrapper from '../page-header-wrapper'

const onClose = vi.fn()

describe('PageHeaderWrapper', () => {
  it('should show banner message', async () => {
    const wrapper = render(
      <PageHeaderWrapper
        banner={<>Banner</>}
        date={<>Date</>}
        tags={<>tags</>}
        utcTime={<>UTC time</>}
        onClickClose={onClose}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})

/**
 * 
 * 
    const mission = {
      status: MissionStatusEnum.ENDED,
      completenessForStats: {
        sources: [MissionSourceEnum.MONITORENV],
        status: CompletenessForStatsStatusEnum.COMPLETE
      }
    } as Mission2

    render(<MissionPageHeader onClickClose={onClose} mission={mission} />)
    waitFor(() => expect(screen.getByText('La mission est terminée et ses données sont complètes')).toBeInTheDocument())
 */
