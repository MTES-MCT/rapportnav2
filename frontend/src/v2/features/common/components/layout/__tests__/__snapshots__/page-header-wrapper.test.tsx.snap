// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeaderWrapper > should show banner message 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-kbeVdv jGFRFQ"
      >
        <div
          class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-space-between"
          style="height: 100%;"
        >
          <div
            class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
          >
            <div
              class="rs-stack"
              style="align-items: center; flex-direction: row; gap: 1rem;"
            >
              <div
                class="rs-stack-item"
              >
                <p
                  class="sc-evkzZa sc-iRcyzz iHjymB dcaSpz"
                  color="#E5E5EB"
                  font-size="22"
                  weight="bold"
                >
                  Date
                </p>
              </div>
              <div
                class="rs-stack-item"
              >
                <div
                  class="sc-lcItFd gA-deIP Element-TagGroup"
                >
                  tags
                </div>
              </div>
            </div>
          </div>
          <div
            class="rs-flex-box-grid-item rs-flex-box-grid-item-6"
          >
            <div
              class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-end"
              style="height: 100%;"
            >
              <div
                class="rs-stack"
                style="align-items: center; flex-direction: row; gap: 2rem;"
              >
                <div
                  class="rs-stack-item"
                >
                  UTC time
                </div>
                <div
                  class="rs-stack-item"
                >
                  <button
                    class="sc-dNdcvo bDxZrn Element-IconButton"
                    role="quit-mission-cross"
                    style="margin-top: 4px;"
                    type="button"
                  >
                    <span
                      class="sc-hjsuWn hZNHWy Element-IconBox"
                    >
                      <svg
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <g
                          transform="translate(160 -82)"
                        >
                          <g
                            fill="none"
                            stroke-miterlimit="10"
                          >
                            <path
                              d="M-142,85.415-143.415,84-150,90.585-156.585,84-158,85.415-151.415,92-158,98.585l1.415,1.415L-150,93.415l6.585,6.585L-142,98.585-148.585,92Z"
                              stroke="none"
                            />
                            <path
                              d="M -156.5853271484375 84 L -150 90.5853271484375 L -143.4146728515625 84 L -142 85.4146728515625 L -148.5853271484375 92 L -142 98.5853271484375 L -143.4146728515625 100 L -150 93.4146728515625 L -156.5853271484375 100 L -158 98.5853271484375 L -151.4146728515625 92 L -158 85.4146728515625 L -156.5853271484375 84 Z"
                              fill="currentColor"
                              stroke="none"
                            />
                          </g>
                          <rect
                            fill="none"
                            height="20"
                            transform="translate(-160 82)"
                            width="20"
                          />
                        </g>
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      Banner
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-kbeVdv jGFRFQ"
    >
      <div
        class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-space-between"
        style="height: 100%;"
      >
        <div
          class="rs-flex-box-grid-item rs-flex-box-grid-item-0"
        >
          <div
            class="rs-stack"
            style="align-items: center; flex-direction: row; gap: 1rem;"
          >
            <div
              class="rs-stack-item"
            >
              <p
                class="sc-evkzZa sc-iRcyzz iHjymB dcaSpz"
                color="#E5E5EB"
                font-size="22"
                weight="bold"
              >
                Date
              </p>
            </div>
            <div
              class="rs-stack-item"
            >
              <div
                class="sc-lcItFd gA-deIP Element-TagGroup"
              >
                tags
              </div>
            </div>
          </div>
        </div>
        <div
          class="rs-flex-box-grid-item rs-flex-box-grid-item-6"
        >
          <div
            class="rs-flex-box-grid rs-flex-box-grid-middle rs-flex-box-grid-end"
            style="height: 100%;"
          >
            <div
              class="rs-stack"
              style="align-items: center; flex-direction: row; gap: 2rem;"
            >
              <div
                class="rs-stack-item"
              >
                UTC time
              </div>
              <div
                class="rs-stack-item"
              >
                <button
                  class="sc-dNdcvo bDxZrn Element-IconButton"
                  role="quit-mission-cross"
                  style="margin-top: 4px;"
                  type="button"
                >
                  <span
                    class="sc-hjsuWn hZNHWy Element-IconBox"
                  >
                    <svg
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <g
                        transform="translate(160 -82)"
                      >
                        <g
                          fill="none"
                          stroke-miterlimit="10"
                        >
                          <path
                            d="M-142,85.415-143.415,84-150,90.585-156.585,84-158,85.415-151.415,92-158,98.585l1.415,1.415L-150,93.415l6.585,6.585L-142,98.585-148.585,92Z"
                            stroke="none"
                          />
                          <path
                            d="M -156.5853271484375 84 L -150 90.5853271484375 L -143.4146728515625 84 L -142 85.4146728515625 L -148.5853271484375 92 L -142 98.5853271484375 L -143.4146728515625 100 L -150 93.4146728515625 L -156.5853271484375 100 L -158 98.5853271484375 L -151.4146728515625 92 L -158 85.4146728515625 L -156.5853271484375 84 Z"
                            fill="currentColor"
                            stroke="none"
                          />
                        </g>
                        <rect
                          fill="none"
                          height="20"
                          transform="translate(-160 82)"
                          width="20"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    Banner
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
